### PR TITLE
Tweak Basic Info Cluster for consistency

### PR DESCRIPTION
**Pull request by lzgrablic02**
#### Problem
* Basic Info Cluster XML inconsistencies

#### Change overview
Tweak Basic Info Cluster XML for consistency with spec and other clusters.
ZAP regen.

#### Testing
* None

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
Changes look ok, but unsure about the cause of the build error: `static_assert(256 < NumericAttributeTraits<uint8_t>::kNullValue, "value.size() might be too big");`

File: src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml [hunk 1]
```diff
@@ -15,34 +15,36 @@ See the License for the specific language governing permissions and
...
+    <attribute side="server" code="8"  define="HARDWARE_VERSION_STRING"   type="char_string" minLength="1" length="64"                                              >HardwareVersionString</attribute>
+    <attribute side="server" code="9"  define="SOFTWARE_VERSION"          type="int32u"                                 default="0"                                 >SoftwareVersion</attribute>
+    <attribute side="server" code="10" define="SOFTWARE_VERSION_STRING"   type="char_string" minLength="1" length="64"                                              >SoftwareVersionString</attribute>
+    <attribute side="server" code="11" define="MANUFACTURING_DATE"        type="char_string" minLength="8" length="16"                               optional="true">ManufacturingDate</attribute>
+    <attribute side="server" code="12" define="PART_NUMBER"               type="char_string"               length="32"                               optional="true">PartNumber</attribute>
+    <attribute side="server" code="13" define="PRODUCT_URL"               type="char_string"               length="256"                              optional="true">ProductURL</attribute>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml:42]: I think char string is at most 254 due to ember limitation. That would explain the "long char string" before (to support 256).

Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml:42]: Yes, you can't do a char_string with a length > 254.  If that compiles, it's a bug that we should fix.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml:42]: Oh, good, we do have compile errors for this.  Way later than they should be (zap gen should fail!), but it's something.

Comment (lzgrablic02) [reply src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml]: Changed back to long

Comment (lzgrablic02) [reply src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml]: Changed back to long

**Comment by bzbarsky-apple** [comment]
The endpoint_config.h changes look like a bad merge?  Please run:
```
  git submodule update --init
```
and then regenerate.

**Comment by lzgrablic02** [comment]
@bzbarsky-apple Ah, yes, sorry, thanks for catching that.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: looks like alignment/whitespace/non-generating field update. There is a default change however zap codegen does not seem affected and the XML looks reasonable.

**Pull request merged**